### PR TITLE
Letterbox implementation

### PR DESCRIPTION
**Pull request by OlfatProgressus**

### CODE REVIEW

File: src/Utils.php [hunk 1]
```diff
@@ -588,15 +597,167 @@ public static function generate_shipping_options_html( $backend_data ) {
...
+	}
+
+	/**
+	 * Check if current order/cart is eligible for automatically use letterbox.
+	 *
+	 * @param \WC_Order|\WC_Cart|int $order \WC_order, \WC_Cart or Order ID.
+	 *
+	 * @return boolean
+	 */
+	public static function is_eligible_auto_letterbox( $order ) {
```
Comment (Abdalsalaam) [inline src/Utils.php:682]: I suggest to create a **single responsibility** function for checking the order and another for checking the cart


```suggestion
	public static function is_order_eligible_auto_letterbox( $order ) {
```

```suggestion
	public static function is_cart_eligible_auto_letterbox( $cart ) {
```

File: src/Utils.php [hunk 2]
```diff
@@ -588,15 +597,167 @@ public static function generate_shipping_options_html( $backend_data ) {
...
+
+		// Check order
+		if ( is_int( $order ) ) {
+			$order = wc_get_order( $order );
+		}
+		if(wc_get_base_location()['country'] == 'BE'){
+			return false;
+		}
```
Comment (Abdalsalaam) [inline src/Utils.php:690]:
```suggestion
		if ( 'BE' !== wc_get_base_location()['country'] ) {
			return false;
		}
```
[When doing logical comparisons involving variables, always put the variable on the right side and put constants, literals, or function calls on the left side](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#yoda-conditions)

File: src/Utils.php [hunk 3]
```diff
@@ -588,15 +597,167 @@ public static function generate_shipping_options_html( $backend_data ) {
...
+			}
+
+			$quantity                    = $item['quantity'] ?? $item->get_quantity();
+			$qty_per_letterbox           = intval( $product->get_meta( Product\Single::MAX_QTY_PER_LETTERBOX ) );
+			$ratio_letterbox_item        = 0 != $qty_per_letterbox ? 1 / $qty_per_letterbox : 0;
+			$total_ratio_letterbox_item += ( $ratio_letterbox_item * $quantity );
+		}
+
+		// If the total ratio is more than 1, that means order items cannot be packed using letterbox.
+		return ( $total_ratio_letterbox_item <= 1 ) ? true : false;
```
Comment (Abdalsalaam) [inline src/Utils.php:748]:
```suggestion
		return $total_ratio_letterbox_item <= 1;
```

**Review by Abdalsalaam — CHANGES_REQUESTED** [review]
Nice work @OlfatProgressus 
I still didn't test the plugin functionality but I left some comments

File: languages/postnlnew.po
[the remark this answers is not in the document]
Comment (OlfatProgressus) [reply languages/postnlnew.po:1]: This file is the new version of translations

**Comment by Abdalsalaam** [comment]
Can you please add the change log ( fixes/features ) ? @olfat123

**Pull request merged**